Permission denied for /usr/etc/hedera/application.yml

### Description

When running the mirror node with docker compose all the images configured to run with the ubuntu user (1000) get an error during startup, i.e.:

```shell
importer-1  | 2025-04-21T18:46:10.237Z ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application run failed
importer-1  | 
importer-1  | java.lang.IllegalStateException: java.io.FileNotFoundException: /usr/etc/hedera/application.yml (Permission denied)
importer-1  |   at org.springframework.beans.factory.config.YamlProcessor.handleProcessError(YamlProcessor.java:226) ~[spring-beans-6.2.5.jar:6.2.5]
importer-1  |   at org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:218) ~[spring-beans-6.2.5.jar:6.2.5]
importer-1  |   at org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:169) ~[spring-beans-6.2.5.jar:6.2.5]
importer-1  |   at org.springframework.boot.env.OriginTrackedYamlLoader.load(OriginTrackedYamlLoader.java:84) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.env.YamlPropertySourceLoader.load(YamlPropertySourceLoader.java:50) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.StandardConfigDataLoader.load(StandardConfigDataLoader.java:54) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.StandardConfigDataLoader.load(StandardConfigDataLoader.java:36) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.ConfigDataLoaders.load(ConfigDataLoaders.java:96) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.ConfigDataImporter.load(ConfigDataImporter.java:132) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.ConfigDataImporter.resolveAndLoad(ConfigDataImporter.java:87) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.ConfigDataEnvironmentContributors.withProcessedImports(ConfigDataEnvironmentContributors.java:121) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.ConfigDataEnvironment.processInitial(ConfigDataEnvironment.java:245) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.ConfigDataEnvironment.processAndApply(ConfigDataEnvironment.java:232) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.ConfigDataEnvironmentPostProcessor.postProcessEnvironment(ConfigDataEnvironmentPostProcessor.java:96) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.config.ConfigDataEnvironmentPostProcessor.postProcessEnvironment(ConfigDataEnvironmentPostProcessor.java:89) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.env.EnvironmentPostProcessorApplicationListener.onApplicationEnvironmentPreparedEvent(EnvironmentPostProcessorApplicationListener.java:132) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.env.EnvironmentPostProcessorApplicationListener.onApplicationEvent(EnvironmentPostProcessorApplicationListener.java:115) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:185) ~[spring-context-6.2.5.jar:6.2.5]
importer-1  |   at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:178) ~[spring-context-6.2.5.jar:6.2.5]
importer-1  |   at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:156) ~[spring-context-6.2.5.jar:6.2.5]
importer-1  |   at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:138) ~[spring-context-6.2.5.jar:6.2.5]
importer-1  |   at org.springframework.boot.context.event.EventPublishingRunListener.multicastInitialEvent(EventPublishingRunListener.java:136) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.context.event.EventPublishingRunListener.environmentPrepared(EventPublishingRunListener.java:81) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.SpringApplicationRunListeners.lambda$environmentPrepared$2(SpringApplicationRunListeners.java:64) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at java.base/java.lang.Iterable.forEach(Unknown Source) ~[na:na]
importer-1  |   at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:118) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:112) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.SpringApplicationRunListeners.environmentPrepared(SpringApplicationRunListeners.java:63) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.SpringApplication.prepareEnvironment(SpringApplication.java:353) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:313) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at com.hedera.mirror.importer.ImporterApplication.main(ImporterApplication.java:15) ~[classes/:0.129.0-SNAPSHOT]
importer-1  |   at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[na:na]
importer-1  |   at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
importer-1  |   at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:102) ~[app/:na]
importer-1  |   at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:64) ~[app/:na]
importer-1  |   at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:40) ~[app/:na]
importer-1  | Caused by: java.io.FileNotFoundException: /usr/etc/hedera/application.yml (Permission denied)
importer-1  |   at java.base/java.io.FileInputStream.open0(Native Method) ~[na:na]
importer-1  |   at java.base/java.io.FileInputStream.open(Unknown Source) ~[na:na]
importer-1  |   at java.base/java.io.FileInputStream.<init>(Unknown Source) ~[na:na]
importer-1  |   at java.base/java.io.FileInputStream.<init>(Unknown Source) ~[na:na]
importer-1  |   at java.base/sun.net.www.protocol.file.FileURLConnection.connect(Unknown Source) ~[na:na]
importer-1  |   at java.base/sun.net.www.protocol.file.FileURLConnection.getInputStream(Unknown Source) ~[na:na]
importer-1  |   at org.springframework.core.io.UrlResource.getInputStream(UrlResource.java:233) ~[spring-core-6.2.5.jar:6.2.5]
importer-1  |   at org.springframework.boot.origin.OriginTrackedResource.getInputStream(OriginTrackedResource.java:61) ~[spring-boot-3.4.4.jar:3.4.4]
importer-1  |   at org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:202) ~[spring-beans-6.2.5.jar:6.2.5]
importer-1  |   ... 36 common frames omitted
importer-1  | 
importer-1 exited with code 1
```

This is due to the change in the configuration for docker compose, moving from volumes to inline configuration.

While when using the volumes the configuration file is mounted with user's permissions, with the new inline config configuration the file is created with root permissions, so the user cannot access it.

```shell
-r--r----- 1 root   root    291 Apr 21 20:35 application-via-config.yml
-rw-r--r-- 1 ubuntu ubuntu    5 Apr 21 20:35 application-via-volume.yml
```

A potential solution is to set all the container to run the internal application as root, i.e.:

```yml
  importer:
    user: root
    configs:
      - source: app-config
        target: /usr/etc/hedera/application.yml
    environment:
      HEDERA_MIRROR_IMPORTER_DB_HOST: db
      SPRING_CONFIG_ADDITIONAL_LOCATION: file:/usr/etc/hedera/
      SPRING_DATA_REDIS_HOST: redis
    image: gcr.io/mirrornode/hedera-mirror-importer:0.129.0-SNAPSHOT
    pull_policy: always
    restart: unless-stopped
    tty: true
```

### Steps to reproduce

1. Clone the repo
2. Run `docker compose up`
3. See the error

### Additional context

_No response_

### Hedera network

_No response_

### Version

v0.127.1

### Operating system

Linux

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -32,6 +32,8 @@ services:
     configs:
       - source: app-config
         target: /usr/etc/hedera/application.yml
+        uid: "1000"
+        gid: "1000"
     deploy:
       replicas: 0
     environment:
@@ -48,6 +50,8 @@ services:
     configs:
       - source: app-config
         target: /usr/etc/hedera/application.yml
+        uid: "1000"
+        gid: "1000"
     environment:
       HEDERA_MIRROR_GRPC_DB_HOST: db
       SPRING_CONFIG_ADDITIONAL_LOCATION: file:/usr/etc/hedera/
@@ -63,6 +67,8 @@ services:
     configs:
       - source: app-config
         target: /usr/etc/hedera/application.yml
+        uid: "1000"
+        gid: "1000"
     environment:
       HEDERA_MIRROR_IMPORTER_DB_HOST: db
       SPRING_CONFIG_ADDITIONAL_LOCATION: file:/usr/etc/hedera/
@@ -76,6 +82,8 @@ services:
     configs:
       - source: app-config
         target: /usr/etc/hedera/application.yml
+        uid: "1000"
+        gid: "1000"
     deploy:
       replicas: 0
     environment:
@@ -97,6 +105,8 @@ services:
     configs:
       - source: app-config
         target: /usr/etc/hedera/application.yml
+        uid: "1000"
+        gid: "1000"
     environment:
       CONFIG_PATH: /usr/etc/hedera/
       HEDERA_MIRROR_REST_DB_HOST: db
@@ -112,6 +122,8 @@ services:
     configs:
       - source: app-config
         target: /usr/etc/hedera/application.yml
+        uid: "1000"
+        gid: "1000"
     environment:
       HEDERA_MIRROR_RESTJAVA_DB_HOST: db
       SPRING_CONFIG_ADDITIONAL_LOCATION: file:/usr/etc/hedera/
@@ -126,6 +138,8 @@ services:
     configs:
       - source: app-config
         target: /usr/etc/hedera/application.yml
+        uid: "1000"
+        gid: "1000"
     deploy:
       replicas: 0
     environment:
@@ -142,6 +156,8 @@ services:
     configs:
       - source: app-config
         target: /usr/etc/hedera/application.yml
+        uid: "1000"
+        gid: "1000"
     environment:
       HEDERA_MIRROR_WEB3_DB_HOST: db
       SPRING_CONFIG_ADDITIONAL_LOCATION: file:/usr/etc/hedera/

